.github/workflows/build-docs.yml

```diff
@@ -77,23 +78,36 @@ jobs:
                     git pull --ff-only upstream "refs/pull/$(cat merge.txt)";
                 fi
 
-      # Set up environment
+        # Set up environment
         -   name: Install apt packages
             run: |
-                ./build_tools/github/build_docs_apt_dependencies.sh
-        -   name: Setup conda
-            uses: conda-incubator/setup-miniconda@v2
+                sudo -E apt-get -yq update
+                sudo -E apt-get -yq --no-install-suggests --no-install-recommends install \
+                    dvipng texlive-latex-base texlive-latex-extra
+        -   name: Setup python
+            uses: actions/setup-python@v4
             with:
-                auto-activate-base: true
-                activate-environment: ''
-                miniconda-version: latest
-                channels: conda-forge
-        -   name: Install packages in conda env
+                python-version: '3.12'
+        # Install the local version of the library, along with both standard and testing-related dependencies
+        # The `doc` dependency group is included because the build_docs job uses this script.
+        # See pyproject.toml for dependency group options
+        -   name: Install packages
+            run: |
+                python -m pip install --user --upgrade pip setuptools
+                python -m pip install .[plotting,doc]
+
+        # Check if we are doing a full or partial build
+        -   name: Find build type
+            run: ./build_tools/github/build_type.sh
+            env:
+                COMMIT_SHA: ${{ github.event.pull_request.head.sha }}
+        -   name: Verify build type
             run: |
-                ./build_tools/github/build_docs_dependencies.sh
+                echo "PATTERN = $(cat pattern.txt)"
+                echo "BUILD = $(cat build.txt)"
```

Comment:
@ymzayek this is after this step you mention we can skip the cache restore if we are on a partial build?

Reply:
Yes so we can skip the caching steps if `BUILD` is set to `html-noplot`